homeassistant/components/versasense/__init__.py

```diff
@@ -87,7 +87,7 @@ def _add_entity_info_to_list(peripheral, device, entity_info_list):
             KEY_PARENT_MAC: device.mac,
         }
 
-        entity_info_list.append(entity_info)
+        entity_info_list[peripheral.identifier] = entity_info
```

Comment:
If we combine the mac address, the peripherial identifierand the measurement name we can guarantee a unique key in the dict. Just the identifier risks duplicates without knowing more about this integration. The entity unique ids uses the same pattern.